core/modules/taxonomy/taxonomy.admin.inc

```diff
@@ -738,27 +738,24 @@ function taxonomy_form_term($form, &$form_state, TaxonomyTerm $term = NULL, Taxo
   // During initial form build, add the term entity to the form state for use
   // during form building and processing. During a rebuild, use what is in the
   // form state.
+  if (!isset($vocabulary) && isset($term->vocabulary)) {
+    $vocabulary = taxonomy_vocabulary_load($term->vocabulary);
+  }
   if (!isset($form_state['term'])) {
```

Comment:
I think the `if` should go above the "During initial form build..." comment, with its own comment above it saying something like: "Load the vocabulary from the term, if not already provided." And then put a blank line between this `if` function and the "During initial form build..." comment.

```suggestion
  // Load the vocabulary from the term, if not already provided.
  if (!isset($vocabulary) && isset($term->vocabulary)) {
    $vocabulary = taxonomy_vocabulary_load($term->vocabulary);
  }
  
  // During initial form build, add the term entity to the form state for use
  // during form building and processing. During a rebuild, use what is in the
  // form state.
  if (!isset($form_state['term'])) {
```